specification/storage/resource-manager/Microsoft.Storage/stable/2018-11-01/storage.json

```diff
@@ -0,0 +1,2317 @@
+        },
+        "encryption": {
+          "$ref": "#/definitions/Encryption",
+          "description": "Provides the encryption settings on the account. The default setting is unencrypted."
+        },
+        "accessTier": {
+          "type": "string",
+          "description": "Required for storage accounts where kind = BlobStorage. The access tier used for billing.",
+          "enum": [
+            "Hot",
+            "Cool"
+          ],
+          "x-ms-enum": {
+            "name": "AccessTier",
+            "modelAsString": false
+          }
+        },
+        "azureFilesAadIntegration": {
+          "type": "boolean",
+          "x-ms-client-name": "EnableAzureFilesAadIntegration",
+          "description": "Enables Azure Files AAD Integration for SMB if sets to true."
+        },
+        "supportsHttpsTrafficOnly": {
+          "type": "boolean",
+          "x-ms-client-name": "EnableHttpsTrafficOnly",
+          "description": "Allows https traffic only to storage service if sets to true."
+        },
+        "networkAcls": {
+          "description": "Network rule set",
+          "x-ms-client-name": "NetworkRuleSet",
+          "$ref": "#/definitions/NetworkRuleSet"
+        }
+      },
+      "description": "The parameters used when updating a storage account."
+    },
+    "StorageAccountUpdateParameters": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "Gets or sets the SKU name. Note that the SKU name cannot be updated to Standard_ZRS, Premium_LRS or Premium_ZRS, nor can accounts of those SKU names be updated to any other value."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key no greater in length than 128 characters and a value no greater in length than 256 characters."
+        },
+        "identity": {
+          "$ref": "#/definitions/Identity",
+          "description": "The identity of the resource."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/StorageAccountPropertiesUpdateParameters",
+          "description": "The parameters used when updating a storage account."
+        },
+        "kind": {
+          "type": "string",
+          "description": "Optional. Indicates the type of storage account. Currently only StorageV2 value supported by server.",
+          "enum": [
+            "Storage",
+            "StorageV2",
+            "BlobStorage",
+            "FileStorage",
+            "BlockBlobStorage"
+          ],
+          "x-ms-enum"
```

Comment:
Thanks for the feedback. As this would require a server change and we are on a tight release schedule, we prefer keeping it as is for now. We will keep in mind to use string enum in the future for flexibility.

Reply:
#Resolved